providers/azureblob/src/main/java/org/jclouds/azureblob/blobstore/AzureBlobRequestSigner.java

```diff
@@ -167,5 +175,71 @@ private HttpRequest sign(String method, String container, String name, @Nullable
       String signature = auth.calculateSignature(stringToSign);
       request.addQueryParam("sig", signature);
       return request.build();
+   
+   }
+   
+   /** 
+   * method, compatible with SAS Authentication 
+   * 
+   */
+   private HttpRequest signSAS(String method, String container, String name, @Nullable GetOptions options, long expires, @Nullable Long contentLength, @Nullable String contentType) {
+      checkNotNull(method, "method");
+      checkNotNull(container, "container");
+      checkNotNull(name, "name");
+	  
+      String nowString = timeStampProvider.get();
+      Date now = dateService.rfc1123DateParse(nowString);
+      Date expiration = new Date(now.getTime() + TimeUnit.SECONDS.toMillis(expires));
+      String iso8601 = dateService.iso8601SecondsDateFormat(expiration);
+      String signedPermission;
+      if (method.equals("PUT")) {
+         signedPermission = "w";
+      } else if (method.equals("DELETE")) {
+         signedPermission = "d";
+      } else {
+         signedPermission = "r";
+      }
+	
+      HttpRequest.Builder request = HttpRequest.builder()
+            .method(method)
+            .endpoint(Uris.uriBuilder(storageUrl).appendPath(container).appendPath(name).query(this.credential).build())
+            .replaceHeader(HttpHeaders.DATE, nowString);
+      
+
+      if (contentLength != null) {
+         request.replaceHeader(HttpHeaders.CONTENT_LENGTH, contentLength.toString());
+      }
+
+      if (contentType != null) {
+         request.replaceHeader("x-ms-blob-content-type", contentType);
+      }
+
+      if (options != null) {
+         request.headers(options.buildRequestHeaders());
+      }
+		
+		 
+      if (method.equals("PUT")) {
+         request.replaceHeader("x-ms-blob-type", "BlockBlob");
+      }
+	 
+	  HttpRequest req = request.build();
+	  
+      return req;
+   }  
```

Comment:
Most of this method is just a copy of the existing one. Break the original one in reusable methods and reuse code.

Reply:
Done!